src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -303,6 +301,147 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
+            For i As Integer = 0 To numElements - 1
+                Dim argumentSyntax As SimpleArgumentSyntax = arguments(i)
+                Dim name As String = Nothing
+
+                Dim nameSyntax As IdentifierNameSyntax = argumentSyntax.NameColonEquals?.Name
+
+                If nameSyntax IsNot Nothing Then
+                    name = nameSyntax.Identifier.ValueText
+                    elementLocations.Add(nameSyntax.GetLocation)
+
+                    countOfExplicitNames += 1
+                    If Not CheckTupleMemberName(name, i, argumentSyntax.NameColonEquals.Name, diagnostics, uniqueFieldNames) Then
+                        hasErrors = True
+                    End If
+                Else
+                    elementLocations.Add(argumentSyntax.GetLocation)
+                End If
+
+                CollectTupleFieldMemberNames(name, i + 1, numElements, elementNames)
+
+                Dim boundArgument As BoundExpression = BindRValue(argumentSyntax.Expression, diagnostics)
+                boundArguments.Add(boundArgument)
+
+                Dim elementType = GetTupleFieldType(boundArgument, argumentSyntax, diagnostics, hasErrors)
+                elementTypes.Add(elementType)
+
+                If elementType Is Nothing Then
+                    hasNaturalType = False
+                End If
+            Next
+
+            uniqueFieldNames.Free()
+
+            If countOfExplicitNames <> 0 AndAlso countOfExplicitNames <> elementTypes.Count Then
+                hasErrors = True
+                ReportDiagnostic(diagnostics, node, ERRID.ERR_TupleExplicitNamesOnAllMembersOrNone)
+            End If
+
+            Dim elementNamesArray = If(elementNames Is Nothing, Nothing, elementNames.ToImmutableAndFree())
+
+            Dim tupleTypeOpt As NamedTypeSymbol = Nothing
+            Dim elements = elementTypes.ToImmutableAndFree()
+            Dim locations = elementLocations.ToImmutableAndFree()
+
+            If hasNaturalType Then
+                tupleTypeOpt = TupleTypeSymbol.Create(node.GetLocation, elements, locations, elementNamesArray, Me.Compilation, node, diagnostics)
+            Else
+                TupleTypeSymbol.VerifyTupleTypePresent(elements.Length, node, Me.Compilation, diagnostics)
+            End If
+
+            Return New BoundTupleLiteral(node, elementNamesArray, boundArguments.ToImmutableAndFree(), tupleTypeOpt, hasErrors)
+        End Function
+
+        ''' <summary>
+        ''' Returns the type to be used as a field type generates errors in case the type is not
+        ''' supported for tuple type fields.
+        ''' </summary>
+        Private Function GetTupleFieldType(expression As BoundExpression, errorSyntax As VisualBasicSyntaxNode, diagnostics As DiagnosticBag, <Out> ByRef hasError As Boolean) As TypeSymbol
+            Dim expressionType As TypeSymbol = expression.Type
+
+            If Not expression.HasErrors Then
+                If expression.Type IsNot Nothing Then
```

Comment:
fixed

---
In reply to: [71440943](https://github.com/dotnet/roslyn/pull/12596#discussion_r71440943) [](ancestors = 71440943)